packages/@vue/cli-plugin-typescript/generator/template-vue3/src/shims-vue.d.ts

```diff
@@ -1,5 +1,5 @@
 declare module '*.vue' {
-  import { defineComponent } from 'vue'
-  const component: ReturnType<typeof defineComponent>
+  import type { DefineComponent } from 'vue'
+  const component: DefineComponent
```

Comment:
@pikax @cexbrayat 
Just to make sure I get it right.
Is this the recommended way to shim `.vue` files now in Vue 3 stable?

Reply:
I think you have something weird with `import type`, but otherwise yes this is what I've been using:

```
declare module '*.vue' {
  import { DefineComponent } from 'vue';
  const component: DefineComponent;
  export default component;
}
```

Reply:
What does "something weird with `import type`" mean?

Reply:
Sorry if I wasn't clear: I don't think `import type` is necessary here, as `DefineComponent` is just a type, so there is no tree-shaking issue. I'm fine if you prefer to keep it, I was just pointing out that the syntax is a bit more advanced, and might lose some developers not familiar with the latest TS features

Reply:
OK. Now that it's only a style issue then I'll keep it. Because now Vue CLI scaffolds with TS 3.9+, which supports this syntax. And the `import type` syntax is quite self-explaining so that people won't mistakenly take `DefineComponent` as a value, making the code a little clearer.